### PR TITLE
Remove oneRealm arg from set_permission

### PR DESCRIPTION
**Pull request by marcoscaceres**
The use of realm and "oneRealm" are being removed from the Permission spec. 
https://github.com/w3c/permissions/pull/397

And related discussion:
https://github.com/w3c/permissions/issues/387

### CODE REVIEW

File: mediacapture-streams/permission-helper.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
 async function setMediaPermission(status="granted", scope=["camera", "microphone"]) {
   try {
     for (let s of scope) {
-      await test_driver.set_permission({ name: s }, status, true);
```
Comment (rakuco) [inline mediacapture-streams/permission-helper.js:7]: From w3c/permissions#387 I took it that WebKit does support this parameter (unlike Gecko and Blink). Would this cause any kind of regression?

File: speculation-rules/prerender/restriction-background-sync.tentative.https.html [hunk 2]
```diff
@@ -30,8 +30,7 @@
   // periodicSync.register() waits until the permission is granted, which
   // is deferred during prerendering so the test would trivially pass without
   // the permission.
-  await test_driver.set_permission({name: 'periodic-background-sync'}, 'granted',
-                                    location.origin, location.origin);
```
Comment (rakuco) [inline speculation-rules/prerender/restriction-background-sync.tentative.https.html:34]: Was this test relying on the first `location.origin` being converted to a boolean? won't removing the second one change the test's behavior?

File: speculation-rules/prerender/restriction-midi-sysex.https.html [hunk 3]
```diff
@@ -21,7 +21,7 @@
   // is deferred during prerendering so the test would trivially pass without
   // the permission.
   await test_driver.set_permission(
-      {name: 'midi', sysex: true}, 'granted', true);
```
Comment (rakuco) [inline speculation-rules/prerender/restriction-midi-sysex.https.html:24]: The speculation-rules/ test all seem to set `one_realm` to true; my question about whether this will cause any regressions in WebKit also applies.

File: webdriver/tests/permissions/set.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ def query(session, name):
     { "descriptor": "geolocation", "state": "granted" },
     { "descriptor": [ { "name": "geolocation" } ], "state": "granted" },
     [ { "descriptor": { "name": "geolocation" }, "state": "granted" } ],
-    { "descriptor": { "name": "geolocation" }, "state": "granted", "oneRealm": 23 }
+    { "descriptor": { "name": "geolocation" }, "state": "granted" }
```
Comment (rakuco) [inline webdriver/tests/permissions/set.py:31]: I think this line can be removed now.

File: webdriver/tests/permissions/set.py [hunk 5]
```diff
@@ -54,11 +54,6 @@ def test_non_secure_context(session, url, state):
     assert_error(response, "invalid argument")
 
 @pytest.mark.parametrize("state", ["granted", "denied", "prompt"])
-@pytest.mark.parametrize("realmSetting", [
```
Comment (rakuco) [inline webdriver/tests/permissions/set.py:57]: I'm not familiar with this part of the code base, but I wonder if any of the tests below can be simplified/removed now that there's no realm handling?

File: mediacapture-streams/permission-helper.js (hunk 1 above)
Comment (marcoscaceres) [reply mediacapture-streams/permission-helper.js:7]: Nope. Should be fine.

File: speculation-rules/prerender/restriction-background-sync.tentative.https.html (hunk 2 above)
Comment (marcoscaceres) [reply speculation-rules/prerender/restriction-background-sync.tentative.https.html:34]: Argument is invalid

File: speculation-rules/prerender/restriction-midi-sysex.https.html (hunk 3 above)
Comment (marcoscaceres) [reply speculation-rules/prerender/restriction-midi-sysex.https.html:24]: Nope. It shouldn’t

File: webdriver/tests/permissions/set.py (hunk 5 above)
Comment (marcoscaceres) [reply webdriver/tests/permissions/set.py:57]: I don't think so... none of them did anything useful in practice as no one actually implemented this.

File: webdriver/tests/permissions/set.py (hunk 4 above)
Comment (marcoscaceres) [inline webdriver/tests/permissions/set.py:31]:
```suggestion
```

Comment (marcoscaceres) [reply webdriver/tests/permissions/set.py]: good catch. Indeed.

File: webdriver/tests/permissions/set.py (hunk 5 above)
Comment (rakuco) [reply webdriver/tests/permissions/set.py:57]: OK. The test failures being reported here look real though; I think you also need to remove `realmSettings` from the test functions for them to pass again.

Comment (marcoscaceres) [reply webdriver/tests/permissions/set.py:57]: Ok, thanks for checking that. Will do!

**Review by reillyeon — APPROVED** [review]
Test changes that I would probably be considered an owner of look good to me. Didn't look at any of the automation infrastructure changes.

File: infrastructure/metadata/infrastructure/testdriver/set_permission.https.html.ini [hunk 6]
```diff
@@ -1,8 +1,8 @@
 [set_permission.https.html]
```
Comment (rakuco) [inline infrastructure/metadata/infrastructure/testdriver/set_permission.https.html.ini:1]: Does removing this file cause Firefox and/or Safari to break? I thought everyone implemented set_permission at this point.

Comment (whimboo) [reply infrastructure/metadata/infrastructure/testdriver/set_permission.https.html.ini:1]: If that's related to https://wpt.fyi/results/webdriver/tests/permissions?label=master&label=experimental&product=chrome&product=firefox&product=safari it's not implemented in Firefox and Safari still.

Comment (rakuco) [reply infrastructure/metadata/infrastructure/testdriver/set_permission.https.html.ini:1]: It's related to https://wpt.fyi/results/infrastructure/testdriver/set_permission.https.html?label=master&label=experimental&product=chrome&product=firefox&product=safari, but in any case it does look like the root cause is the same, so thanks for pointing it out :-)

**Review by rakuco — APPROVED** [review]
lgtm

The Chromium failure in the checks looks unrelated; I'd try pushing again to see if it goes away, otherwise someone with enough power will have to force-merge this PR.

File: infrastructure/testdriver/set_permission.https.html [hunk 7]
```diff
@@ -1,17 +1,23 @@
 <!DOCTYPE html>
-<meta charset="utf-8">
+<meta charset="utf-8" />
```
Comment (annevk) [inline infrastructure/testdriver/set_permission.https.html:2]: This seems unrelated?

File: infrastructure/testdriver/set_permission.https.html [hunk 8]
```diff
@@ -1,17 +1,23 @@
...
+        await test_driver.set_permission(descriptor, "granted");
+        permission = await navigator.permissions.query(descriptor);
+        assert_equals(permission.state, "granted");
+    }, "Grant Permission");
+
+    promise_test(async (t) => {
+        await test_driver.set_permission(descriptor, "denied");
+        permission = await navigator.permissions.query(descriptor);
+        assert_equals(permission.state, "denied");
+    }, "Deny Permission");
```
Comment (annevk) [inline infrastructure/testdriver/set_permission.https.html:22]: Should we really be changing this entire test as part of this?

**Review by annevk — COMMENTED** [review]
I missed this was already being reviewed by @rakuco. Anyway, couple small drive-by nits.

File: infrastructure/testdriver/set_permission.https.html (hunk 7 above)
Comment (marcoscaceres) [reply infrastructure/testdriver/set_permission.https.html:2]: No, ran prettier over it… it adds the “ />” garbage.

File: infrastructure/testdriver/set_permission.https.html (hunk 8 above)
Comment (marcoscaceres) [reply infrastructure/testdriver/set_permission.https.html:22]: In this case, yes. It doesn’t do any harm to test that setting the permission worked.
